### DISCUSSION TITLE
Some example for: "redux-toolkit"+"persist"+"typescript"

### DISCUSSION THREAD

**Discussion by yardz** [vercel/next.js#21939] [Ideas]
**Describe the feature you'd like to request**

I would like an example using the "redux-toolkit" + "persist" if possible with typescript.
I started a project using "with-typescript-eslint-jest" as a base (https://github.com/vercel/next.js/tree/canary/examples/with-typescript-eslint-jest).

but I need to put redux with persist in the project... The installation of redux with toolkit went smoothly, but when I tried to add persist I couldn't.

**Describe the solution you'd like**

just an example with redux toolkit + typescript + persist.
There is actually no example of redux with typescript or any example of "persist + toolkit"

**Describe alternatives you've considered**

I tried to use the toolkit and then adapt the persist setup but I couldn't

**Comment by MaliugaSergiy** [comment]
I also would like to combine next + redux-toolkit + persist.
Does anyone have an example template?

**Reply by yardz** [reply]
In my case the states that I need to maintain are very simple, and can be loaded only on the client...
So I did the following. I installed the toolkit normally, but added the "persistor"... This was getting in the way of SSG generating the HTML correctly, so I just removed the "PersistGate" from "_app", and work normally.

In the case of my application, the persisted data is for a few pages (only 2), so I handle these states manually...

This is absolutely not the "right way" ... but it worked.
If you use the "PersistGate" the application works, but the SSG delivers a "bad html" to be indexed, so I prefer not to use it because my application needs a lot of good SEO (main reason to leave CRA and come to the Nextjs).

**Reply by Carlosf96** [reply]
> In my case the states that I need to maintain are very simple, and can be loaded only on the client...
> So I did the following. I installed the toolkit normally, but added the "persistor"... This was getting in the way of SSG generating the HTML correctly, so I just removed the "PersistGate" from "_app", and work normally.
> 
> In the case of my application, the persisted data is for a few pages (only 2), so I handle these states manually...
> 
> This is absolutely not the "right way" ... but it worked.
> If you use the "PersistGate" the application works, but the SSG delivers a "bad html" to be indexed, so I prefer not to use it because my application needs a lot of good SEO (main reason to leave CRA and come to the Nextjs).

Hey bro could you perhaps upload that as an example in a gist or something pls?¡

**Reply by yardz** [reply]
My _app.tsx:
```ts
export default function MyApp({ Component, ...props }) {
	const router = useRouter();
	const compact = ['/produto/[ref]'].includes(router.route);
	return (
		<Provider store={store}>
				<Template categories={categories || []} compact={compact}>
					<Component {...props} />
				</Template>
				<ToastContainer />
		</Provider>
	);
}
```

I really just didn't put PersistGate. the rest is the same. Again, it works for my context as persist doesn't change my public pages at all.

The only bad side effect is that when the user reloads a private page (and in these the persistence can change a little) sometimes it takes hundredths of seconds to get the information on the screen... But it's an extremely low price for my context.
